libs/secure-sm/src/main/java/module-info.java

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+module org.elasticsearch.secure_sm {
```

Comment:
Naming is hard, but a completely fair (and in fact good) point to raise here. We should strive for something reasonable and consistent (where possible). 

Simple names typically make for better module names (rather than compound words), but we should be judicious in our use of dots too since they are commonly used in Java to represent logical hierarchies. For secure-sm, it seems best to just compound it to `securesm`, since it is unlikely that there will be further modules in the logical group `o.e.secure`. 

I've taken a complete pass over all the names, now we have:
```
$ find . -name "module-info.java" -exec grep -e "^module" {} \; | sort
module org.elasticsearch.aggs.matrix {
module org.elasticsearch.analysis.icu {
module org.elasticsearch.base {
module org.elasticsearch.cli {
module org.elasticsearch.geo {
module org.elasticsearch.lz4 {
module org.elasticsearch.mapper.unsignedlong {
module org.elasticsearch.painless {
module org.elasticsearch.painless.spi {
module org.elasticsearch.percolator {
module org.elasticsearch.pluginclassloader {
module org.elasticsearch.rankeval {
module org.elasticsearch.securesm {
module org.elasticsearch.security {
module org.elasticsearch.server {
module org.elasticsearch.sslconfig {
module org.elasticsearch.transport.netty4 {
module org.elasticsearch.xcontent {
module org.elasticsearch.xcontent.impl {
module org.elasticsearch.xcore {
```